### PR TITLE
add language  to charge request

### PR DESCRIPTION
**Pull request by AhmedAliIbrahim**

### CODE REVIEW

**Comment by AhmedAliIbrahim** [comment]
@amrrbakry  please review

File: lib/fawry/contracts/charge_request_contract.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@ class ChargeRequestContract < Dry::Validation::Contract
         required(:amount).value(:decimal)
         required(:description).value(:string)
         required(:customer_mobile).value(:string)
+        required(:language).value(:string)
```
Comment (amrrbakry) [inline lib/fawry/contracts/charge_request_contract.rb:14]: Thank you for another great contribution! 🤝 did Fawry add this required param recently? it's in their integration documentation, but I don't remember seeing it before and it breaks backward compatibility 🤔

Comment (AhmedAliIbrahim) [reply lib/fawry/contracts/charge_request_contract.rb:14]: it's my mistake :D, it's required on their docs, i saw it first time last week, but if we didn't send it it works well

image


I'll update the pr later today to make it optional and add a default value

Comment (AhmedAliIbrahim) [reply lib/fawry/contracts/charge_request_contract.rb]: @amrrbakry updated!

Comment (amrrbakry) [reply lib/fawry/contracts/charge_request_contract.rb]: no problem. their documentation is shit 😂 yes, I think that would be a good idea to make it optional and also make sure the value is either `"ar-eg"` or `"en-gb"` to match the documentation:

```ruby
optional(:language).value(included_in?: %w[ar-eg en-gb])
```
and we add a default value in the charge request default params here:

https://github.com/fawry-api/fawry/blob/0ed3abb9c1a4270e0f1e16d467008fc41524a1c7/lib/fawry/requests/charge_request.rb#L8

something like that 😄 that way we don't break backward compatibility too 👍 

Thank you so much 🤝 I will add you as a collaborator after merging this change, and I trust you will maintain this project well 💯

File: lib/fawry/requests/charge_request.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@
 module Fawry
   module Requests
     module ChargeRequest
-      DEFAULTS = { payment_method: 'PAYATFAWRY', currency_code: 'EGP' }.freeze
+      DEFAULTS = { payment_method: 'PAYATFAWRY', currency_code: 'EGP', language: 'en-gb' }.freeze
```
Comment (amrrbakry) [inline lib/fawry/requests/charge_request.rb:8]: I think it's better to have it as `'ar-eg'` 'cause most people want to get the notification in Arabic. WDYT?

Comment (AhmedAliIbrahim) [reply lib/fawry/requests/charge_request.rb:8]: Makes sense

Comment (AhmedAliIbrahim) [reply lib/fawry/requests/charge_request.rb]: updated!

**Review by amrrbakry — APPROVED** [review]

**Comment by amrrbakry** [comment]
@AhmedAliIbrahim released to RubyGems! thanks 🤝

**Pull request merged**